src/drive/mobile/modules/settings/duck/reducers.js

```diff
@@ -4,18 +4,22 @@ import {
   BACKUP_IMAGES,
   BACKUP_CONTACTS,
   SET_ANALYTICS,
-  WIFI_ONLY
+  WIFI_ONLY,
+  ADD_MEDIA_BUCKET,
+  DEL_MEDIA_BUCKET
 } from './actions'
 
 const initialState = {
   serverUrl: '',
   offline: false,
   analytics: false,
   backupImages: false,
-  wifiOnly: true
+  wifiOnly: true,
+  mediaBuckets: new Set(['Camera'])
 }
 
 const settings = (state = initialState, action) => {
+  state = { ...initialState, ...state }
```

Comment:
@y-lohse Do you have an idea of why @lmeunier has to do that to have access to its `mediaBuckets`? Do we persist `settings` store?

Reply:
Yes we do that [here](https://github.com/cozy/cozy-drive/blob/master/src/drive/store/configureStore.js#L39).

We also [have this](https://github.com/cozy/cozy-drive/blob/master/src/drive/store/persistedState.js#L30) still hanging around, but it shouldn't be triggered anymore.

I'm not sure it's useful to keep this line though, the reducer will simply add the `mediaBuckets` entry when it's needed?

Reply:
So to persist this new Settings, we have to add it here : https://github.com/cozy/cozy-drive/blob/master/src/drive/store/configureStore.js#L39 

With this modification, even with a kill / reboot of the app we will still have the buckets ;)

Reply:
I don't think that's needed? [This line](https://github.com/cozy/cozy-drive/blob/master/src/drive/store/configureStore.js#L42) already saves the complete settings. So once the buckets get in the state, they should be automatically saved. But they won't be there on the first boot.

I also wonder if using a `Set` will be a problem, not sure it will be properly deserialized.

Reply:
This line is needed when `state.mobile.settings` exists but does not have the `mediaBuckets` entry. In this case, [getSetting(state, "mediaBucket")](https://github.com/cozy/cozy-drive/blob/master/src/drive/mobile/modules/settings/duck/reducers.js#L39) always returns `undefined` (instead of the default `Set(['Camera'])`) and an exception is thrown on the "Select Media Buckets" page.
This is typically the case when a user is already using the mobile app (`state.mobile.settings` exists) and the app is updated with this feature. Without this line, `state.mobile.settings.mediaBuckets` would still be undefined.

Reply:
So to do the things right, we should migrate the persisted state to add this default value, right? 

This default value should only target `Android`. We can create a `addDefaultMediaBucket` migration (https://github.com/cozy/cozy-drive/blob/master/src/drive/store/persistedState.js#L59) to add this default value. 

WDYT?

Reply:
Would work, or we could do something like:

```
export const getMediaBuckets = state => getSetting(state, 'mediaBucket') || new Set(['Camera']))
```

… over here : https://github.com/cozy/cozy-drive/blob/master/src/drive/mobile/modules/settings/duck/reducers.js#L54

I don't really mind which one :)

Reply:
If I'm not wrong, we have 3 choices:
1. keep the line `state = { ...initialState, ...state }`
2. create a `addDefaultMediaBucket` migration to add the default value
3. add something like `export const getMediaBuckets = state => getSetting(state, 'mediaBucket') || new Set(['Camera']))`

So, which one should be implemented?

Reply:
I'm not comfortable with option 1 because we're not supposed to mutate the current `state`. I think 2 is a bit more complicated than 3 and it seems ike we'll forget removing the migration afterwards… so I would go for option 3. @Crash-- you've looked at this PR more than me, WDYT?

Reply:
3 sounds good ;)

Reply:
Fixed in the last force-push:
https://github.com/cozy/cozy-drive/compare/1e292de87acc3b74bee2e22a6334dceea6ab257a..ecb6127e4b63984bc9b569e11c8fc0834167a7dd